libs/platform/src/lib/components/list/list.component.ts

```diff
@@ -0,0 +1,657 @@
+export type ListType = 'inactive' | 'active' | 'detail';
+let nextListId = 0;
+let nextListGrpHeaderId = 0;
+export type FdpListDataSource<T> = ListDataSource<T> | T[];
+export class SelectionChangeEvent {
+    selectedItems: BaseListItem[];
+    index: number;
+}
+
+/**
+ * The List component represents a container for list item types.
+ * It is used to display a list features.
+ */
+@Component({
+    selector: 'fdp-list',
+    templateUrl: './list.component.html',
+    encapsulation: ViewEncapsulation.None,
+    styleUrls: ['./list.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [{ provide: FormFieldControl, useExisting: ListComponent, multi: true }]
+})
+
+
+export class ListComponent extends CollectionBaseInput implements OnInit, AfterViewInit, AfterContentInit, OnDestroy {
+
+    /**Enables lazy loadMore of data */
+    @Input()
+    loadMore: boolean;
+
+    /**Enables data load on scroll for true
+     * false: enables data loading on button
+     * click
+     */
+    @Input()
+    loadOnScroll: boolean;
+
+    /**Wait time for new items */
+    @Input()
+    delayTime: number;
+
+    /**Max height for viewport to display scroll */
+    @Input()
+    maxHeight: string;
+
+    /**Items to be loaded at once */
+    @Input()
+    itemSize: number;
+
+    /**Scroll offset percentage to trigger scroll event */
+    @Input()
+    scrollOffsetPercentage: number;
+
+    /** Whether list component has removed borders */
+    @Input()
+    noBorder = false;
+
+    @Input()
+    selection = false;
+
+    /** Whether list component has removed bottom borders */
+    @Input()
+    noSeperator: boolean;
+
+    /** The type of the selection. Types include:
+    *'none'| 'multi' | 'single'|'delete'.
+    * Leave empty for default ().'
+    * Default value is set to ''
+    */
+    @Input()
+    selectionMode: SelectionType = 'none';
+
+    /**ListType 'inactive' | 'active' | 'navigation' | 'detail' */
+    @Input()
+    listType: ListType = 'active';
+
+    /**  An array that holds a list of all selected items**/
+    @Input()
+    selectedItems: BaseListItem[];
+
+    /** define size of items for screen reader */
+    @Input()
+    ariaSetsize: number;
+
+    /** Defines whether items are multiseletable for screen reader */
+    @Input()
+    ariaMultiselectable: boolean;
+
+    /**Title used on button when data loads on button click */
+    @Input()
+    loadTitle: string;
+
+    /**
+   * Todo: Name of the entity for which DataProvider will be loaded.
+   *  You can either pass list of
+   * items or use this entityClass and internally we should be able
+   * to do lookup to some registry
+   * and retrieve the best matching DataProvider that is set on application level
+   *
+   *
+   */
+    @Input()
+    entityClass: string;
+
+
+    /**
+      * Child items of the List.
+      */
+    @ContentChildren(BaseListItem, { descendants: true }) ListItems: QueryList<BaseListItem>;
```

Comment:
Let's be consistent, everywhere you have new lines when using decorator  - here you have it all in one line

Reply:
done